src/Cello/problem_ScalarExpr.cpp

```diff
@@ -70,6 +53,18 @@ void ScalarExpr::evaluate (T * value, double t,
 	  ndx,ndy,ndz,nx,ny,nz,
 	  (ndx >= nx) && (ndy >= ny) && (ndz >= nz));
 
+  if ((!param_) && (!mask)) {
+    // fast-path that avoid heap allocations - relevant for inflow conditions
+    for (int ix=0; ix<nx; ix++) {
+      for (int iy=0; iy<ny; iy++) {
+	for (int iz=0; iz<nz; iz++) {
+          value[ix + ndx*(iy + ndy*iz)] = (T)value_;
```

Comment:
Since arrays are stored using column-major ordering I try to order loops with iz on the outside and ix on the inside. This leads to more sequential memory accesses, which can improve cache performance.

Reply:
Fixed - I though it was weird that I ordered the loop in this way, so I looked around.

It turns out that there were some other pre-existing code in that file that ordered loops in this way. I fixed those loops too.